File: BettysCandyCane.java
import java.io.*;
import java.util.ArrayList;

public class Main {
  public static void main(String[] args) throws IOException {
    BufferedReader userIn = new BufferedReader(new InputStreamReader (System.in));
    int defVal = Integer.MAX_VALUE;
    int startX = 0;
    int startY = 0;
		String[] tokens = userIn.readLine().split(" ");
    int endX = Integer.parseInt(tokens[0]);
    int endY = Integer.parseInt(tokens[1]);
    int [][] dist = new int [10][10];
    char [][] grid = {
      {'B', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', 'X'},
      {' ', ' ', ' ', ' ', ' ', ' ', 'X', ' ', ' ', ' '},
      {' ', ' ', 'X', ' ', ' ', ' ', ' ', ' ', 'X', ' '},
      {'X', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '},
      {' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '},
      {'X', 'X', 'X', 'X', 'X', 'X', 'X', ' ', ' ', ' '},
      {' ', ' ', ' ', ' ', ' ', ' ', ' ', 'X', ' ', ' '},
      {' ', 'X', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '},
      {' ', ' ', ' ', 'X', ' ', ' ', ' ', 'X', ' ', ' '},
      {' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', 'X'}
    };
    for (int i = 0; i < 10; i++) {
      for (int j = 0; j < 10; j++) {
        dist [i][j] = defVal;
        if (grid [i][j] == 'B') {
          startX = j;
          startY = i;
        }
        else if (grid [i][j] == 'X') {
          dist [i][j] = -1;
        }

      }
    }
    ArrayList <Integer> xQueue = new ArrayList <Integer>();
    ArrayList <Integer> yQueue = new ArrayList <Integer>();
    xQueue.add(startX);
    yQueue.add(startY);
    dist [startY][startX] = 0;
    while (!xQueue.isEmpty()) {
      int x = xQueue.get(0);
      xQueue.remove(0);
      int y = yQueue.get(0);
      yQueue.remove(0);
      if (grid [y][x] == ' ' || grid [y][x] == 'B') {
        if (y + 1 < 10) {
          if (dist [y + 1][x] != -1 && dist [y + 1][x] == defVal) {
            dist[y + 1][x] = dist [y][x] + 1;
            xQueue.add(x);
            yQueue.add(y + 1);
          }
        }
        if (x + 1 < 10) {
          if (dist [y][x + 1] != -1 && dist [y][x + 1] == defVal) {
            dist[y][x + 1] = dist [y][x] + 1;
            xQueue.add(x + 1);
            yQueue.add(y);
          }
        }
      if (grid [y][x] == ' ' || grid [y][x] == 'B') {
        if (y - 1 > -1) {
          if (dist [y - 1][x] != -1 && dist [y - 1][x] == defVal) {
            dist[y - 1][x] = dist [y][x] + 1;
            xQueue.add(x);
            yQueue.add(y - 1);
          }
        }
        if (x - 1 > -1) {
          if (dist [y][x - 1] != -1 && dist [y][x - 1] == defVal) {
            dist[y][x - 1] = dist [y][x] + 1;
            xQueue.add(x - 1);
            yQueue.add(y);
          }
        }
      }
    }
  }
  if (dist[endY - 1][endX - 1] == -1) {
    System.out.println("This is a wall.");
  }
  else {
    System.out.println("Betty will sweat " + dist[endY - 1][endX - 1] + "ml to get to her candy cane");
  }
}
}

File: BettysCandyCane.py
import sys
# endX, endY = map(int,input().strip().split(" ")) 
endX, endY = [int(x) for x in input().split()]
grid = [
  ['B', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', 'X'], 
  [' ', ' ', ' ', ' ', ' ', ' ', 'X', ' ', ' ', ' '], 
  [' ', ' ', 'X', ' ', ' ', ' ', ' ', ' ', 'X', ' '], 
  ['X', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '], 
  [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
  ['X', 'X', 'X', 'X', 'X', 'X', 'X', ' ', ' ', ' '],
  [' ', ' ', ' ', ' ', ' ', ' ', ' ', 'X', ' ', ' '],
  [' ', 'X', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
  [' ', ' ', ' ', 'X', ' ', ' ', ' ', 'X', ' ', ' '],
  [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', 'X']
]
defVal = sys.maxsize
dist = [[defVal, defVal, defVal]]
for i in range(0, 10):
  dist.append([])
  for j in range (0, 10):
    dist[i].append(defVal)
    if grid [i][j] == 'B':
      dist [i][j] = 0
      startX = j #only needed if B is a random value
      startY = i
    elif grid [i][j] == 'X':
      dist [i][j] = -1

xQueue = [0]
yQueue = [0]
xQueue.append(startX)
yQueue.append(startY)
while len(xQueue) > 0:
  x = xQueue[0]
  del xQueue[0]
  y = yQueue[0]
  del yQueue[0]
  if grid [y][x] == 'B' or grid [y][x] == ' ':
    if y + 1 < 10:
      if dist [y + 1][x] == defVal:
        dist [y + 1][x] = dist [y][x] + 1
        xQueue.append(x)
        yQueue.append(y + 1)
    if x + 1 < 10:
      if dist [y][x + 1] == defVal:
        dist [y][x + 1] = dist [y][x] + 1
        xQueue.append(x + 1)
        yQueue.append(y)
    if y - 1 > -1:
      if dist [y - 1][x] == defVal:
        dist [y - 1][x] = dist [y][x] + 1
        xQueue.append(x)
        yQueue.append(y - 1)
    if x - 1 > -1:
      if dist [y][x - 1] == defVal:
        dist [y][x - 1] = dist [y][x] + 1
        xQueue.append(x - 1)
        yQueue.append(y)

# for i in range (10):
  # print(dist[i])

print ("Betty will sweat " + str(dist[endY - 1][endX - 1]) + "ml to get to her candy cane")
  

